heron/api/src/java/org/apache/heron/streamlet/Streamlet.java

```diff
@@ -71,6 +71,22 @@
    */
   int getNumPartitions();
 
+  /**
+   * Set the id of the stream to be used by the children nodes.
+   * Usage (assuming source is a Streamlet object with two output streams: stream1 and stream2):
+   *   source.withStream("stream1").filter(...).log();
+   *   source.withStream("stream2").filter(...).log();
+   * @param streamId The specified stream id
+   * @return Returns back the Streamlet with changed stream id
+   */
+  Streamlet<R> withStream(String streamId);
```

Comment:
I think `streamId` offers to be distinguished streams, right? 
Also, wondering is there UI impact (if user would like to see `streamId` or not)?

Reply:
Yeah. A component can generates multiple different streams (such as 3 integer streams: odd, even, all) and the next component needs to specify which one to accept data.

UI doesn't show this information currently and it could be a useful feature to add. And grouping strategy is another important information on the edges.

Reply:
Exactly. Vertices show Streamlet name and Edges can show both streamId and grouping strategy.

Reply:
As long-term, i think we can raise an UI issue to be addressed this (to be shown both **streamId** and **grouping strategy**).

Reply:
Yep.

Reply:
I have just created for this: #3117